### PR TITLE
feat: auto-adding of schema data based on tokens studio data

### PR DESCRIPTION
**Pull request by mrcjhicks**
## Description

Adds schema information to token definitions matching types in the tokens studio data.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by GarthDB** [comment]
## Run report for [5dff6b64](https://github.com/adobe/spectrum-tokens/commit/5dff6b643c66b617be466526285b502a1fb7498d)
Total time: 53.1s | Comparison time: 54.8s | Estimated savings: 1.8s (3.2% faster)
|     | Action | Time | Status | Info |
| :-: | :----- | ---: | :----- | :--- |
| 🟩 | `SyncWorkspace` | 1.3s | Passed |  |
| ⬛️ | `SetupNodeTool(18.13.0)` | 2s | Skipped |  |
| 🟩 | `InstallNodeDeps(18.13.0)` | 7.6s | Passed |  |
| 🟩 | `SyncNodeProject(visualizer)` | 0.2ms | Passed |  |
| 🟩 | `SyncNodeProject(tokens)` | 0.3ms | Passed |  |
| 🟩 | `SyncNodeProject(site)` | 0.2ms | Passed |  |
| 🟩 | `RunTask(site:copySchema)` | 74.3ms | Passed |  |
| 🟩 | `RunTask(visualizer:typescript)` | 2.3s | Passed |  |
| 🟩 | `RunTask(tokens:buildTokens)` | 13.9s | Passed |  |
| 🟩 | `RunTask(tokens:buildManifest)` | 317.1ms | Passed |  |
| 🟩 | `RunTask(visualizer:build)` | 4.2s | Passed |  |
| 🟩 | `RunTask(tokens:build)` | 2.1s | Passed |  |
| 🟩 | `RunTask(tokens:generateDiffResult)` | 4.1s | Passed |  |
| 🟩 | `RunTask(tokens:test)` | 2.1s | Passed |  |
| 🟩 | `RunTask(site:build)` | 23.2s | Passed |  |
| 🟩 | `RunTask(site:clean)` | 122.7ms | Passed |  |
| 🟩 | `RunTask(site:export)` | 814.4ms | Passed |  |

<details><summary><strong>Touched files</strong></summary><div>

```
.changeset/dirty-pets-attend.md
.changeset/good-peas-mix.md
.changeset/happy-students-vanish.md
.changeset/lemon-dragons-raise.md
.changeset/neat-balloons-rhyme.md
.changeset/pre.json
.changeset/slimy-emus-enjoy.md
.changeset/strange-rings-love.md
.changeset/tame-bobcats-beg.md
.changeset/warm-experts-beg.md
.changeset/wise-turtles-confess.md
.github/workflows/deploy-docs.yml
.github/workflows/release.yml
.moon/toolchain.yml
.moon/workspace.yml
package-lock.json
package.json
packages/tokens/CHANGELOG.md
packages/tokens/package.json
packages/tokens/src/color-aliases.json
packages/tokens/src/color-component.json
packages/tokens/src/color-palette.json
packages/tokens/src/icons.json
packages/tokens/src/layout-component.json
packages/tokens/src/layout.json
packages/tokens/src/semantic-color-palette.json
packages/tokens/src/typography.json
packages/tokens/tasks/diff.js
pnpm-lock.yaml
pnpm-workspace.yaml
tools/transform-tokens-json/README.md
tools/transform-tokens-json/moon.yml
tools/transform-tokens-json/package.json
tools/transform-tokens-json/src/lib/shared.js
tools/transform-tokens-json/src/merge-dark-darkest.js
tools/transform-tokens-json/src/merge-spectrum-express.js
tools/transform-tokens-json/test/index.test.js
```

</div></details>

**Comment by GarthDB** [comment]
I think I got all the basic `$schema` issues addressed, but all the new colors are missing `wireframe` options. I'll follow up with design a solution for this so our tests work, but we also don't lose the data for the existing wireframe values.

File: packages/tokens/src/typography.json [hunk 1]
```diff
@@ -1,1359 +1,1644 @@
 {
   "default-font-family": {
+    "$schema": "https://opensource.adobe.com/spectrum-tokens/schemas/token-types/alias.json",
```
Comment (karstens) [inline packages/tokens/src/typography.json:3]: @mrcjhicks @GarthDB Shouldn't this be font-family?

File: packages/tokens/src/color-component.json [hunk 2]
```diff
@@ -1,226 +1,266 @@
...
   },
   "color-handle-outer-border-opacity": {
     "component": "color-handle",
+    "$schema": "https://opensource.adobe.com/spectrum-tokens/schemas/token-types/alias.json",
     "value": "{color-handle-inner-border-opacity}",
     "uuid": "0a0eff89-7e7d-4b7f-bcfa-a85e33e5798a"
   },
   "color-handle-drop-shadow-color": {
     "component": "color-handle",
+    "$schema": "https://opensource.adobe.com/spectrum-tokens/schemas/token-types/alias.json",
```
Comment (karstens) [inline packages/tokens/src/color-component.json:193]: @GarthDB This token does not appear in the FTS data anymore. Is it possible, that this is a bug?

File: packages/tokens/src/color-component.json [hunk 3]
```diff
@@ -1,226 +1,266 @@
...
     "value": "{drop-shadow-color}",
     "uuid": "97dfaf7f-ac2d-4517-84cc-6f692f712fc5"
   },
   "floating-action-button-drop-shadow-color": {
     "component": "floating-action-button",
+    "$schema": "https://opensource.adobe.com/spectrum-tokens/schemas/token-types/alias.json",
     "value": "{transparent-black-300}",
     "uuid": "d66d3a56-1ba0-4ce7-a263-eb46a4d17af0"
   },
   "floating-action-button-shadow-color": {
```
Comment (karstens) [inline packages/tokens/src/color-component.json:205]: @GarthDB This "deprecated" token was just removed in the S2 FTS Data. I will check, if the sync script has synced that already.

File: packages/tokens/src/layout-component.json [hunk 4]
```diff
@@ -1,5208 +1,7048 @@
...
         "uuid": "ceeda4da-026b-496c-8abd-7081aceae262"
       },
       "mobile": {
+        "$schema": "https://opensource.adobe.com/spectrum-tokens/schemas/token-types/dimension.json",
         "value": "240px",
         "uuid": "dc5f4966-4ddd-4e9a-a748-370d8eaae568"
       }
     }
   },
   "meter-default-width": {
```
Comment (karstens) [inline packages/tokens/src/layout-component.json:1474]: @GarthDB This deprecated token is also removed from spectrum2. The sync script will ignore and not update it in spectrum-tokens. So, can you remove it here, please?

Comment (karstens) [reply packages/tokens/src/layout-component.json:1474]: Okay, got it. This token was deprecated _before_ the spectrum 2 was branched. So, the sync script just has nothing to do here. 👍

**Review by karstens — APPROVED** [review]
LGTM!

File: packages/tokens/src/color-component.json (hunk 2 above)
Comment (GarthDB) [reply packages/tokens/src/color-component.json:193]: It's possible, we could remove it but I think we could do that in a different pr.

**Pull request merged**